### PR TITLE
feat(api): support thermocycler commands in executor

### LINKED ISSUE #5500: Thermocycler: Protocol Execution
As a user, I would like to be able to run JSON protocols including TC steps on my OT-2. 

## Acceptance Criteria
- [ ] Robot executor supports JSON protocol commands for Thermocycler:
  - [ ] openLid
  - [ ] closeLid
  - [ ] activateLid
  - [ ] deactivateLid
  - [ ] activateBlock
  - [ ] deactivateBlock
  - [ ] runProfile

## Implementation Notes
Note this is currently implemented as async.

### PR DESCRIPTION
**Pull request by IanLondon**
## overview

Closes #5500

Should support all v4 JSON thermocycler commands.

Includes test fixture `shared-data/protocol/fixtures/4/testThermocyclerModuleV4.json` which contains all the thermocycler commands.

## review requests

- [ ] QA: uploading the test protocol `shared-data/protocol/fixtures/4/testThermocyclerModuleV4.json` should run on a robot using this branch's version of the server
- [ ] Code review, big ol' sanity check
- [ ] Also: closing this issue means we think API server is fully ready for any TC commands we'll throw at it. Does the test protocol encompass all the situations we need to handle (eg if you think about all possible outputs of TC State / TC Profile, do the existing commands implemented as blocking give us the control we need)?

## risk assessment

Medium, robot API

### CODE REVIEW

File: shared-data/protocol/schemas/4.json [hunk 1]
```diff
@@ -65,7 +65,7 @@
     },
 
     "slot": {
-      "description": "Slot on the deck of an OT-2 robot if 1-12. Else, a module ID",
+      "description": "string '1'-'12', or special string 'span7_8_10_11' signify it's a slot on the OT-2 deck. If it's a UUID, it's the slot on the module referenced by that ID.",
```
Comment (IanLondon) [inline shared-data/protocol/schemas/4.json:68]: non-functional description change to schema

**Review by nusrat813 — APPROVED** [review]
Tested  `shared-data/protocol/fixtures/4/testThermocyclerModuleV4.json`

working as intended

**Review by sfoster1 — APPROVED** [review]
LGTM

**Review by Kadee80 — APPROVED** [review]
🌡️ ♻️ 
Don't have a TC but I could upload to my robot.

**Review by shlokamin — APPROVED** [review]
🚢

**Pull request merged**